From bace737dd0ac073deb75fe67481d767669357943 Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Wed, 2 Oct 2024 11:28:03 +0200
Subject: [PATCH 1256/1695] dts: stm32: add ltdc support on stm32mp21

Add ltdc support on stm32mp21 platform.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git d3df15f191b845fbe5cfc5860b30895cf0da6bcd dts: stm32: add ltdc support on stm32mp21]
Change-Id: I4c48a7df55473cc26bfc2a5e7782503790b2cbbc
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/408339
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Yannick FERTRE <yannick.fertre@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Yannick FERTRE <yannick.fertre@foss.st.com>
Tested-by: Yannick FERTRE <yannick.fertre@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/arch/arm/dts/stm32mp211.dtsi |  8 ++++++++
 core/arch/arm/dts/stm32mp215.dtsi | 27 ++++++++++++++++++++++++++-
 2 files changed, 34 insertions(+), 1 deletion(-)

diff --git a/core/arch/arm/dts/stm32mp211.dtsi b/core/arch/arm/dts/stm32mp211.dtsi
index 8b872e297..f4dd1052b 100644
--- a/core/arch/arm/dts/stm32mp211.dtsi
+++ b/core/arch/arm/dts/stm32mp211.dtsi
@@ -2757,4 +2757,12 @@
 				 <4 0 0x80000000 0x10000000>; /* NAND */
 		};
 	};
+
+	soc1: soc@1 {
+		compatible = "simple-bus";
+		#address-cells = <1>;
+		#size-cells = <1>;
+		interrupt-parent = <&intc>;
+		ranges = <0x0 0x0 0x0 0x80000000>;
+	};
 };
diff --git a/core/arch/arm/dts/stm32mp215.dtsi b/core/arch/arm/dts/stm32mp215.dtsi
index 6e9955088..a1495ed41 100644
--- a/core/arch/arm/dts/stm32mp215.dtsi
+++ b/core/arch/arm/dts/stm32mp215.dtsi
@@ -5,5 +5,30 @@
  */
 #include "stm32mp213.dtsi"
 
-/ {
+&soc1 {
+	ltdc: display-controller@48010000 {
+		compatible = "st,stm32-ltdc";
+		reg = <0x48010000 0x400>;
+		interrupts = <GIC_SPI 143 IRQ_TYPE_LEVEL_HIGH>,
+			     <GIC_SPI 144 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&rcc CK_BUS_LTDC>, <&rcc CK_KER_LTDC>;
+		clock-names = "bus", "lcd";
+		resets = <&rcc LTDC_R>;
+		status = "disabled";
+		access-controllers = <&rifsc STM32MP21_RIFSC_LTDC_CMN_ID>;
+		access-controller-names = "cmn";
+
+		l1l2 {
+			access-controllers = <&rifsc STM32MP21_RIFSC_LTDC_L0L1_ID>;
+			access-controller-names = "l1l2";
+		};
+		l3 {
+			access-controllers = <&rifsc STM32MP21_RIFSC_LTDC_L2_ID>;
+			access-controller-names = "l3";
+		};
+		rot {
+			access-controllers = <&rifsc STM32MP21_RIFSC_LTDC_ROT_ID>;
+			access-controller-names = "rot";
+		};
+	};
 };
-- 
2.47.2

